test/SparqlAntlrParserTest.cpp

```diff
@@ -855,3 +855,33 @@ TEST(SparqlParser, RDFLiteral) {
       "\"1950-01-01T00:00:00\"^^<http://www.w3.org/2001/XMLSchema#dateTime>");
   expectRDFLiteralFails(R"(?a ?b "The \"Moon\""@en .)");
 }
+
+TEST(SparqlParser, SelectQuery) {
+  auto expectSelectQuery = ExpectCompleteParse<&Parser::selectQuery>{
+      {{INTERNAL_PREDICATE_PREFIX_NAME, INTERNAL_PREDICATE_PREFIX_IRI}}};
+  auto expectSelectQueryFails = ExpectParseFails<&Parser::selectQuery>{};
+  expectSelectQuery(
+      "SELECT * WHERE { ?a <bar> ?foo }",
+      m::ParsedQuery("SELECT * WHERE { ?a <bar> ?foo }", m::AsteriskSelect(),
+                     m::GraphPattern(m::Triples({{"?a", "<bar>", "?foo"}}))));
+  expectSelectQuery(
+      "SELECT ?foo WHERE { } GROUP BY ?foo",
+      m::ParsedQuery("SELECT ?foo WHERE { } GROUP BY ?foo",
+                     m::VariablesSelect({"?foo"}), m::GraphPattern(), {}, {},
+                     {Variable{"?foo"}}, {}));
```

Comment:
Could you please check the following:
This query is accepted by the parser, but throws then during query planning with a rather strange exception (No suitable query plan found). In this case it would be nicer to directly throw in the parser with `QLever currently doesn't support empty GroupGraphPatterns and WHERE clauses`.

Reply:
Yes it seems that at least one `GraphPatternOperation` (excluding `SparqlFilter`) or `BasicGraphPattern` (Triples) is required. I have added a check.